FoodFriends/app/src/main/java/com/example/foodfriends/activities/LogInActivity.java

```diff
@@ -31,31 +32,47 @@ protected void onCreate(Bundle savedInstanceState) {
         etPassword = findViewById(R.id.etLogInPassword);
         btnLogin = findViewById(R.id.btnLogIn);
         tvSignUp = findViewById(R.id.tvSignUpLink);
-        tvSignUp.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Intent i = new Intent(LogInActivity.this, SignUpActivity.class);
-                startActivity(i);
+        tvSignUp.setOnClickListener(this);
+        tvSignUp.setOnClickListener(this);
+        btnLogin.setOnClickListener(this);
 
-            }
-        });
-        btnLogin.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
+    }
+
+    /**
+     * Provides different functionality depending on view clicked
+     * Either taking user to sign up page or logging the user in**/
+    @Override
+    public void onClick(View v) {
```

Comment:
Hmmm generally we'd create different click listeners for each button rather than rely on a switch

Reply:
When I was following the tutorial for MVC, it said to remove the onclick listeners with code inside them from the oncreate method and out into an overall on click listener, which handles the cases based on the view clicked, so that's why I changed it to this!

Reply:
ahhh okay gotcha!